### ISSUE TITLE
stochastic policy for continuous control

### ISSUE THREAD

**Issue by sufengniu** [MorvanZhou/Reinforcement-learning-with-tensorflow#16]
Hello, I found that in continuous control case (A3C/PPO), the code use neural network to generate Gaussian distribution, then sample it from the distribution, like the following code:
```
log_prob = self.normal_dist.log_prob(self.a)  # loss without advantage
self.exp_v = log_prob * self.td_error  # advantage (TD_error) guided loss
# Add cross entropy cost to encourage exploration
self.exp_v += 0.01*self.normal_dist.entropy()
```
In the tensorflow documents, the normal_dist.log_prob represents the log probability density function (not probability), and therefore the normal_dist.prob could be greater than 1.0, and log_prob could be positive. In that case, exp_v  might be positive (let's assume entropy loss is 0). In that case, the loss become negative value. I know this is most common way for continuous action, but apparently, from above analytics, it seems problematic. I was wondering if you have any insights to explain it? Thank you!

**Comment by MorvanZhou** [comment]
Hi, you are correct about the pdf and the log pdf. When considering continuous action, it is a discrete point in pdf. The pdf may be greater than 1, but I don't think it will affect the result. The sign of loss is not that important in this case as long as we know we maximize the predicted value.

**Comment by MorvanZhou** [comment]
Hi, you are correct about the pdf and the log pdf. When considering continuous action, it is a discrete point in pdf. The pdf may be greater than 1, but I don't think it will affect the result. The sign of loss is not that important in this case as long as we know we maximize the predicted value.
